### ISSUE TITLE
Differentiate trackpad from mouse wheels on web (Windows & Linux)

### ISSUE THREAD

**Issue by Piinks** [flutter/flutter#139990]
With Flutter web on some desktop platforms (Windows and Linux), we do not know the difference between a trackpad and a physical mouse scroll wheel when the input is received by the framework. Because of this, both mouse and trackpad input is treated the same way, going through the pointerScroll method of ScrollPosition to handle the input.

When we know the difference, the trackpad is handled differently, supporting other interactions like panning, zooming, smoothing of input in accordance with the scroll physics upon release, etc. This should be available on all platforms, not just some.

For Mac, a heuristic was developed to tell the difference between these input devices, and has worked really well:
- https://github.com/flutter/engine/pull/36346

We need to do the same for Windows and Linux so that the experience is consistent across all platforms.

This is also needed to resolve https://github.com/flutter/flutter/issues/32120

**Comment by yjbanov** [comment]
The original PR points to these pages:

```
    // https://searchfox.org/mozilla-central/source/dom/events/WheelEvent.h
    // https://learn.microsoft.com/en-us/windows/win32/inputdev/wm-mousewheel
```

So I thought Windows was covered. How severe are issues due to the confusion about the device kind? If the bugs are very severe, I think the framework might hold off on relying on the difference too much.

**Comment by Piinks** [comment]
>  If the bugs are very severe, I think the framework might hold off on relying on the difference too much.

Yup, we are holding off until we can tell the difference. While trying to resolve https://github.com/flutter/flutter/issues/32120 we found the trackpad could not be accurately filtered out of the incoming input so scrolling was erratic.

**Comment by ditman** [comment]
I thought the `deltaMode` would be enough to tell devices apart?

**Comment by Piinks** [comment]
I thought so too. I don't have a windows or linux machine on hand to test, but @loic-sharma did recently for me. This may also be related to https://github.com/flutter/flutter/issues/76913 which resurfaced in my inbox over the holiday.

**Comment by oleksii-pimenov** [comment]
Hey guys, any progress relate to this issue?

**Comment by GabriellArthur** [comment]
provisionally if you enable AssistiveTouch and add Scroll Down and Scroll Up in "Customize Top Level Menu", you will be able to scroll in applications built with Flutter
